tests/conversions.rs

```diff
@@ -118,3 +118,18 @@ fn test_extra_conversions_const() {
     let a = ConstGenericRingBuffer::<_, 2>::from(a);
     assert_eq!(a.to_vec(), vec![1, 2]);
 }
+
+#[test]
+fn test_const_generic_new_parameter() {
+    // Can we specify size only on the method?
+    let mut a = ConstGenericRingBuffer::new::<2>();
+    a.push(5);
+
+    // Can we specify size in both positions?
+    let mut a = ConstGenericRingBuffer::<i32, 50>::new::<50>();
```

Comment:
What exactly happens if they mismatch, compiler error or does one take precedence?

Reply:
It gives a compile error. Sadly, the error is a bit wordy because of the many `From` implementations.
```
or[E0277]: the trait bound `ConstGenericRingBuffer<_, 4>: From<ConstGenericRingBuffer<_, 5>>` is not satisfied
   --> examples/foo.rs:4:57
    |
4   |     let mut buf = ConstGenericRingBuffer::<_, 4>::new::<5>();
    |                                                         ^ the trait `From<ConstGenericRingBuffer<_, 5>>` is not implemented for `ConstGenericRingBuffer<_, 4>`
    |
    = help: the following other types implement trait `From<T>`:
              <ConstGenericRingBuffer<T, CAP> as From<&[T; CAP]>>
              <ConstGenericRingBuffer<T, CAP> as From<&[T]>>
              <ConstGenericRingBuffer<T, CAP> as From<&mut [T; CAP]>>
              <ConstGenericRingBuffer<T, CAP> as From<&mut [T]>>
              <ConstGenericRingBuffer<T, CAP> as From<AllocRingBuffer<T, SIZE>>>
              <ConstGenericRingBuffer<T, CAP> as From<GrowableAllocRingBuffer<T>>>
              <ConstGenericRingBuffer<T, CAP> as From<LinkedList<T>>>
              <ConstGenericRingBuffer<T, CAP> as From<Vec<T>>>
            and 4 others
note: required by a bound in `ConstGenericRingBuffer::<T, CAP>::new`
   --> /home/terts/Programming/ringbuffer/src/with_const_generics.rs:179:41
    |
179 |         ConstGenericRingBuffer<T, CAP>: From<ConstGenericRingBuffer<T, N>>,
    |                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `ConstGenericRingBuffer::<T, CAP>::new`

For more information about this error, try `rustc --explain E0277`.
```